src/dp/rand_bigint.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2025 ISRG
+// SPDX-License-Identifier: MPL-2.0 AND (MIT OR Apache-2.0)
+//
+// This file contains code covered by the following copyright and permission notice
+// and has been modified by ISRG and collaborators.
+//
+// Copyright 2013-2014 The Rust Project Developers.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Helper functions to randomly generate big integers, sampled from uniform distributions.
+//!
+//! These were vendored from num-bigint.
+
+use num_bigint::BigUint;
+use num_integer::Integer;
+use num_traits::Zero;
+use rand::distr::uniform::Error;
+use rand::Rng;
+
+/// Uniform distribution producing [`BigUint`].
+pub(super) struct UniformBigUint {
+    len: BigUint,
+    base: BigUint,
+}
+
+impl UniformBigUint {
+    pub(super) fn new(low: &BigUint, high: &BigUint) -> Result<Self, Error> {
```

Comment:
Since the internal representation of this is `base, len` anyway, I think it'd be better for the arguments here to match that, because then construction would be infallible. However, since we're vendoring this code and already presumably have a bunch of usage of this pattern elsewhere, we should stick with what's already done.